src/views/virtualmachines/details/tabs/configuration/initialrun/utils/utils.ts

```diff
@@ -0,0 +1,97 @@
+import { ConfigMapModel } from '@kubevirt-ui/kubevirt-api/console';
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { IoK8sApiCoreV1ConfigMap } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import {
+  AUTOUNATTEND,
+  generateNewSysprepConfig,
+  SYSPREP,
+  sysprepDisk,
+  sysprepVolume,
+  UNATTEND,
+} from '@kubevirt-utils/components/SysprepModal/sysprep-utils';
+import { getDisks, getVolumes } from '@kubevirt-utils/resources/vm';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { k8sCreate, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+
+export const patchVMWithExistingSysprepConfigMap = async (
+  name: string,
+  vm: V1VirtualMachine,
+): Promise<void> => {
+  const vmVolumes = getVolumes(vm);
+  const vmDisks = getDisks(vm);
+
+  await k8sPatch<V1VirtualMachine>({
+    data: [
+      {
+        op: 'replace',
+        path: `/spec/template/spec/domain/devices/disks`,
+        value: [
+          ...vmDisks.filter((disk) => disk?.name !== SYSPREP),
+          ...(!isEmpty(name) ? [sysprepDisk()] : []),
```

Comment:
Hmm, wouldn't this line 31 work simply without spread operator and not putting the result of the condition into an array? (same for the line 39)

Reply:
I mean why to put `sysprepDisk()` into an array and then to use spread operator to "unpack" that array.. seems to me unnecessary. Simply using `!isEmpty(name) ? sysprepDisk() : null,` in that line didn't work as expected? For me it seems to work. Am I missing something? :D

Reply:
`undefined` neither?

Reply:
Same. It's a value. Its a primitive type.

Reply:
You are right. But.. does it ever matter? Would presence of null or undefined in that array in that place mean anything from the backend perspective? From my experience, sometimes such values are simply ignored or thrown away when sent to backend, and that feels so nice and makes our life easier (but depending if that value has some specific meaning in that specific place or not, depending on the backend, of course, we should not always rely on it, we should know what we do).

Reply:
i dont know didnt test it, but I don't see a reason sending a null or undefined value if I have other options.

Reply:
🆗